lib/mixins/property-effects.js

```diff
@@ -1676,12 +1723,21 @@ export const PropertyEffects = dedupingMixin(superClass => {
       // ----------------------------
       let hasPaths = this.__dataHasPaths;
       this.__dataHasPaths = false;
+      let notifyProps;
       // Compute properties
-      runComputedEffects(this, changedProps, oldProps, hasPaths);
-      // Clear notify properties prior to possible reentry (propagate, observe),
-      // but after computing effects have a chance to add to them
-      let notifyProps = this.__dataToNotify;
-      this.__dataToNotify = null;
+      if (legacyNoBatch) {
+        // When not batching, computed effects may re-enter, so capture
+        // notifying properties early
+        notifyProps = this.__dataToNotify;
+        this.__dataToNotify = null;
+        runEffects(this, this[TYPES.COMPUTE], changedProps, oldProps, hasPaths);
```

Comment:
Add a comment about why we're not using `runComputedEffects` here.

Reply:
Done